### PR TITLE
Add node initialization & integration test

### PR DESCRIPTION
**Pull request by junha1**

### CODE REVIEW

File: node/src/node.rs [hunk 1]
```diff
@@ -13,8 +15,93 @@ pub struct Node<N: GossipNetwork, S: Storage, R: RawRepository> {
...
+            peers.clone(),
+        )
+        .await?;
+        let governance = Governance::new(dms, Some(config.private_key.clone())).await?;
+
+        // Step 3: initialize the consensus module
+        StorageImpl::create(&format!("{}/consensus/dms", path))
+            .await
+            .unwrap();
+        let storage = StorageImpl::open(path).await.unwrap();
```
Comment (lemonholic) [inline node/src/node.rs:80]: It should be
`let storage = StorageImpl::open(&format!("{}/consensus/dms", path)).await.unwrap();`

File: node/src/node.rs [hunk 2]
```diff
@@ -13,8 +15,93 @@ pub struct Node<N: GossipNetwork, S: Storage, R: RawRepository> {
...
+            storage,
+            consensus_dms_key,
+            dms_config.clone(),
+            peers.clone(),
+        )
+        .await?;
+        StorageImpl::create(&format!("{}/consensus/state", path))
+            .await
+            .unwrap();
+        let consensus_state_storage = StorageImpl::open(path).await.unwrap();
```
Comment (lemonholic) [inline node/src/node.rs:91]: ditto

File: node/src/node.rs [hunk 3]
```diff
@@ -13,8 +15,93 @@ pub struct Node<N: GossipNetwork, S: Storage, R: RawRepository> {
...
+            private_key: config.private_key.clone(),
+        };
+        let dms_config = dms::Config {
+            fetch_interval: Some(std::time::Duration::from_millis(500)),
+            broadcast_interval: Some(std::time::Duration::from_millis(500)),
+            network_config,
+        };
+
+        // Step 2: initialize the governance module
+        StorageImpl::create(&format!("{}/governance/dms", path))
```
Comment (lemonholic) [inline node/src/node.rs:63]: It should be
let storage = StorageImpl::open(&format!("{}/governance/dms", path)).await.unwrap();

**Review by lemonholic — APPROVED** [review]

File: DIRECTORY.md [hunk 4]
```diff
@@ -3,19 +3,24 @@
...
 ## `/consensus`
 
-- `dms/`: DMS for consensus module
\ No newline at end of file
+- `dms/`: DMS for the consensus module
+- `state/`: the state of the consensus module
+
+## `/peers`
+
+various file for trakcing peers in the network
```
Comment (TomTaehoonKim) [inline DIRECTORY.md:26]:
```suggestion
various files for tracking peers in the network
```

File: node/Cargo.toml [hunk 5]
```diff
@@ -20,3 +20,7 @@ simperby-consensus = { version = "0.0.0", path = "../consensus" }
 simperby-repository = { version = "0.0.0", path = "../repository" }
 thiserror = "1.0.32"
 semver = "1.0.0"
+
+[dev-dependencies]
+rand = "0.8.5"
+simperby-test-suite = { path = "../test-suite" }
```
Comment (TomTaehoonKim) [inline node/Cargo.toml:26]:
```suggestion
simperby-test-suite = { path = "../test-suite" }

```

**Review by YoonhoKim0527 — APPROVED** [review]
approved.

File: governance/src/lib.rs [hunk 6]
```diff
@@ -8,6 +8,13 @@ use std::collections::{HashMap, HashSet};
 
 pub type Error = eyre::Error;
 
+pub fn generate_dms_key(header: &BlockHeader) -> String {
+    format!(
+        "governance-{}-{}",
+        header.height,
+        &header.to_hash256().to_string()[0..8]
```
Comment (TomTaehoonKim) [inline governance/src/lib.rs:15]: how about using a constant here?

**Review by TomTaehoonKim — APPROVED** [review]

**Pull request merged**